File: python/3033.py
from typing import List

class Solution:
    def modifiedMatrix(self, matrix: List[List[int]]) -> List[List[int]]:
        n = len(matrix[0])
        max_col = [0] * len(n)
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                max_col[j] = max(max_col[j], matrix[i][j])
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if matrix[i][j] == -1:
                    matrix[i][j] = max_col[j]
        return matrix

File: go/3033.go
package main

func modifiedMatrix(matrix [][]int) [][]int {
	n := len(matrix[0])
	max_col := make([]int, n)
	for i := range matrix {
		for j := range matrix[0] {
			max_col[j] = max(matrix[i][j], max_col[j])
		}
	}
	for i := range matrix {
		for j := range matrix[0] {
			if matrix[i][j] == -1 {
				matrix[i][j] = max_col[j]
			}
		}
	}
	return matrix
}
